dandischema/metadata.py

```diff
@@ -311,13 +311,16 @@ def _add_asset_to_stats(assetmeta: Dict[str, Any], stats: _stats_type) -> None:
             stats = _get_samples(value, stats, hierarchy)
             break
 
+    # which components already found, so we do not count more than
+    # once in some incorrectly named datasets
+    found: Dict[str, str] = {}
     for part in Path(assetmeta["path"]).name.split(".")[0].split("_"):
-        if part.startswith("sub-"):
-            subject = part.replace("sub-", "")
+        if found.get("subject") and part.startswith("sub-"):
```

Comment:
shouldn't this also be `not found...` like `sample` below.

Reply:
rright! so
```suggestion
        if not found.get("subject") and part.startswith("sub-"):
```